### PR TITLE
Remove CountryTranslation and use Intl Component.

### PR DESCRIPTION
**Pull request by gperdomor**
| Q             | A
| ------------- | ---
| Bug fix?      | [yes]
| New feature?  | [yes]
| BC breaks?    | [no]
| Deprecations? | [no]
| Fixed tickets | [#2710]
| License       | MIT
| Doc PR        | [-]

Hi... This PR solves problems of Country model like i'm describe in #2710.

In resume, the country translations are removed to avoid the creation of invalid countries like countries that not really exist, or countries with incorrect name or incorrect iso name, for example, a country with name Venezuela and iso name US.

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Perfect, could you have a look at the failing build? Also, you should not modify translation files other than English language. The extra keys should be removed on sync. Thanks!

**Comment by gperdomor** [comment]
@pjedrzejewski i never worked with behat or phpSpec... I tried to fix the test, but i still get this error: <code>[Semantical Error] line 0, col 66 near 'name = :name': Error: Class Sylius\Component\Addressing\Model\Country has no field or association named name (Doctrine\ORM\Query\QueryException)</code>... What i need to change to solve that?

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,5 @@ bin/*
 !bin/install
 !bin/sylius-cs-fixer
 vendor
+
```
Comment (aRn0D) [inline .gitignore]: you need to revert taht !

File: src/Sylius/Bundle/AddressingBundle/Resources/translations/messages.af.yml [hunk 2]
```diff
@@ -14,7 +14,6 @@ sylius:
       province: Provinsie
     country:
       iso_name: ISO Land kode
-      name: Naam
```
Comment (aRn0D) [inline src/Sylius/Bundle/AddressingBundle/Resources/translations/messages.af.yml]: You just need to remove the translation from the english file :)

File: src/Sylius/Component/Addressing/Model/Country.php [hunk 3]
```diff
@@ -29,14 +30,12 @@ class Country extends AbstractTranslatable implements CountryInterface
      * @var mixed
      */
     protected $id;
-
```
Comment (aRn0D) [inline src/Sylius/Component/Addressing/Model/Country.php]: You need to revert this change.

File: src/Sylius/Component/Addressing/Model/Country.php [hunk 4]
```diff
@@ -29,14 +30,12 @@ class Country extends AbstractTranslatable implements CountryInterface
...
     /**
      * Country name in ISO format.
      *
      * @var string
      */
     protected $isoName;
-
```
Comment (aRn0D) [inline src/Sylius/Component/Addressing/Model/Country.php]: You need to revert this change.

File: src/Sylius/Component/Addressing/Model/Country.php [hunk 5]
```diff
@@ -44,40 +43,41 @@ class Country extends AbstractTranslatable implements CountryInterface
...
     {
-        return $this->translate()->getName();
+        return $this->provinces->contains($province);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function setName($name)
+    public function getId()
```
Comment (aRn0D) [inline src/Sylius/Component/Addressing/Model/Country.php]: You need to put this method at the top of the class

File: src/Sylius/Component/Addressing/Model/Country.php [hunk 6]
```diff
@@ -44,40 +43,41 @@ class Country extends AbstractTranslatable implements CountryInterface
 
     public function __construct()
     {
-        parent::__construct();
-
         $this->provinces = new ArrayCollection();
     }
 
     public function __toString()
```
Comment (aRn0D) [inline src/Sylius/Component/Addressing/Model/Country.php]: You need to keep the same order for the methods

**Comment by aRn0D** [comment]
@gperdomor the country model has not `name` property anymore but only `isoName`, you should update the doctrine mapping and display the name of the country by using the iso code.

**Comment by aRn0D** [comment]
You need to update CountryTypeSpec too !

**Comment by pjedrzejewski** [comment]
@gperdomor Friendly ping. :)

**Comment by gperdomor** [comment]
@pjedrzejewski @aRn0D sorry for the delay... I updated the code :)

**Comment by michalmarcinkowski** [comment]
@gperdomor There are still scenarios that don't pass.

**Comment by gperdomor** [comment]
@michalmarcinkowski yes... the error is the same that i comment previously on may 9... The test are calling findOneByName function in the DefaultContext of the ResourceBundle, which is extended by the AddressingContext on the AddressingBundle, i override that method but for some reason does not work, i dont know why :(... This is my first time working with behat so im sure that i missing something hehe

**Comment by gperdomor** [comment]
ping @pjedrzejewski... can you review this and give me some feedback or help about the error... :D

**Comment by pamil** [comment]
@gperdomor I took care of these errors (#3035), thanks for contributing :)

**Comment by gperdomor** [comment]
Close in favor of #3035... Thanks to @pamil for continue this PR...
